### PR TITLE
Add a vLLM attention route for Mistral

### PR DESCRIPTION
**Pull request by anthony-etim**
## Description of the change

Adds a vLLM attention route to Mistral, so `CachedMistralAttention` can serve through vLLM's TPU backend on the native flax/nnx path, following the routes already in #2794 for GPT-2, Llama, Qwen, and Gemma.

In `keras_hub/src/models/mistral/mistral_attention.py`, `call()` delegates to a new `_compute_vllm_attention` while a serving context is active and runs the original dense path otherwise. The route projects Q/K/V, applies RoPE at the engine's per-token positions, and calls the shared paged-attention bridge, forwarding Mistral's sliding window. The layer's `cache` argument comes back untouched: while serving, vLLM owns the paged KV cache, so the dense one is bypassed.

The kernel expands K/V for grouped-query attention itself, so the route passes the unexpanded `num_key_value_heads`.

Mistral's sliding window is applied by the decoder's mask in the dense path. While serving there is no dense mask, so the window goes to the kernel instead.

## Tests

`keras_hub/src/vllm/mistral_route_test.py` drives a real `CachedMistralAttention` with a recording stand-in kernel, so it runs on CPU with no TPU or vLLM install:

- the route passes the unexpanded K/V head count, the right query head count, and the layer's own scale
- the sliding window reaches the kernel, and is `None` when the layer has none
- off the serving path the layer's output is unchanged and the kernel is never called

Runs on the jax, tensorflow, and torch backends.

## Reference

Depends on keras-team/keras-hub#2794, which adds `keras_hub/src/vllm/` and the bridge this route calls; that should merge first. Companion backend PR: vllm-project/tpu-inference#3104.

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends.
- [x] My PR is based on the latest changes of the main branch.
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md).
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md).
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

File: keras_hub/src/models/mistral/mistral_attention.py [hunk 1]
```diff
@@ -191,6 +204,37 @@ def _compute_key_value(x):
...
+    def _compute_vllm_attention(self, hidden_states, cache, vllm_context):
+        """vLLM paged-attention route (serving on TPU only).
+
+        Projects Q/K/V, applies RoPE at the engine's per-token positions,
+        and hands the (unexpanded) K/V to the shared paged-attention bridge,
+        forwarding Mistral's sliding window. The kernel handles
+        grouped-query attention via ``num_kv_heads``, so the dense path's
+        `ops.repeat` of K/V is not needed here.
+        """
+        positions = ops.reshape(vllm_context.positions, (-1, 1))
```
[the remark this answers is not in the document]
Comment (anthony-etim) [reply keras_hub/src/models/mistral/mistral_attention.py:216]: Not needed, vLLM packs tokens flat so seq_len is always 1 here and both reshapes give the same shape.

**Comment by divyashreepathihalli** [comment]
@jules Can you verify the changes in this PR. Is this backward compatible and is teh attention logic still preserved?

**Review by divyashreepathihalli — APPROVED** [review]
